src/pdfz.cpp

```diff
@@ -432,6 +558,25 @@ HEMI_KERNEL(eval_pdf)(int npoints, const int* read_bins,
 
 
 void EvalHist::EvalAsync(bool do_eval_pdf) {
+
+  // Create a random number generator to pass to oscillation systematic if needed.
```

Comment:
Does this need to happen for every call of `EvalHist::EvalAsync`, or just once e.g. in the `EvalHist` constructor?